controller/Lists.php

```diff
@@ -77,19 +70,18 @@ class Lists extends tao_actions_CommonModule
      * - Returns the page with the lists for GET requests
      * - Creates a new list and returns its name and URI for POST requests
      *
-     * @return void
-     *
-     * @throws common_exception_BadRequest
+     * @throws common_Exception
+     * @throws common_ext_ExtensionException
+     * @throws core_kernel_persistence_Exception
      */
-    public function index()
+    public function index(): void
     {
         if ($this->getPsrRequest()->getMethod() === 'POST') {
             if (!$this->isXmlHttpRequest()) {
                 throw new common_exception_BadRequest('wrong request mode');
             }
```

Comment:
Suggestion: This check is repeated in 8 different methods, maybe we can add a helper method `assertIsXmlHttpRequest()` in this controller and turn all of them into oneliners (`$this->assertIsXmlHttpRequest()` instead of the whole `if` block).

Reply:
Done 🙂